### PR TITLE
feat (client): driver for tauri sqlite

### PR DESCRIPTION
**Pull request by kevin-dp**

### CODE REVIEW

File: clients/typescript/package.json [hunk 1]
```diff
@@ -177,6 +181,7 @@
   "devDependencies": {
     "@electric-sql/prisma-generator": "workspace:*",
     "@ikscodes/browser-env": "^1.0.0",
+    "@tauri-apps/plugin-sql": "2.0.0-alpha.5",
```
Comment (msfstef) [inline clients/typescript/package.json:184]: This is added as both a peer and dev dependency, and not marked as optional - is this intended?

File: clients/typescript/src/drivers/tauri-sqlite/index.ts
Comment (msfstef) [inline clients/typescript/src/drivers/tauri-sqlite/index.ts:1]: nit: there's no driver mock and test files that we have for other drivers - they don't test much but might be worth adding them at least for the logic of name assignment and valid instantiation?

**Review by msfstef — APPROVED** [review]
Looks good!

File: clients/typescript/package.json (hunk 1 above)
Comment (samwillis) [reply clients/typescript/package.json:184]: I believe it is. All the drivers are required as devDependencies for development and in peerDependencies to show that they can be optional installed in a project when support is required.

File: clients/typescript/src/drivers/tauri-sqlite/index.ts
Comment (samwillis) [reply clients/typescript/src/drivers/tauri-sqlite/index.ts:1]: I don't think it's an issue with this driver for now. We are going to add some e2e tests for the drivers at some point.

**Review by samwillis — APPROVED** [review]
LGTM

**Pull request merged**